### ISSUE TITLE
Stat size, parsed size...?

### ISSUE THREAD

**Issue by jmeas** [th0r/webpack-bundle-analyzer#61]
Are there definitions somewhere of exactly what "stat size", "parsed size," etc. means? I'd be happy to add these definitions to the README if they're somewhere else right now. Just lmk!

Thanks for the great plugin ✌️

**Comment by valscion** [comment]
Stat sizes are fetched from the webpack stats object directly, and use the actual source code of your modules as-is and report the size before minification or gzip

Parsed size is calculated by reading the *actual compiled bundle files* and making a link back to the modules from webpack stats file. So if you use a minifier, such as UglifyJS, the parsed size shows you the size after minification.

Likewise, the gzip size is calculated by reading the actual compiled bundle files *and running gzip for each one of the module sources separately*. Thus the gzip size shows you the size after minification _and_ gzip, but it isn't a 1-to-1 mapping with the actual file sizes as gzipping each module separately yields less "wins" in terms of compression as the separate sources have less opportunities for gzip to compress together.

Here's the relevant two lines in the code, that is ran for each module.

https://github.com/th0r/webpack-bundle-analyzer/blob/v2.3.1/src/analyzer.js#L80-L83

It would be nice if you would be able to put this together into the README somehow :). This question has been asked a few times already, so it would make sense to display it in the README.

**Comment by th0r** [comment]
> Thus the gzip size shows you the size after minification and gzip, but it isn't a 1-to-1 mapping with the actual file sizes as gzipping each module separately yields less "wins" in terms of compression as the separate sources have less opportunities for gzip to compress together.

It's not correct. It gzips [each module separately](https://github.com/th0r/webpack-bundle-analyzer/blob/v2.3.1/src/analyzer.js#L71) **but** also gzips [the whole bundles](https://github.com/th0r/webpack-bundle-analyzer/blob/v2.3.1/src/analyzer.js#L82) so you will have relevant gzip sizes for your bundle scripts.

**Comment by valscion** [comment]
Yeah true, forgot to say that for the entire bundle, the gzip size is correct, but for individual modules it might lie. Thanks for correcting me!

**Comment by th0r** [comment]
@valscion Thanks for your answer)

**Comment by jmeas** [comment]
This is immensely helpful. Thanks for all of this detail!

I'll whip up a PR to add this to the README shortly.

**Comment by jmeas** [comment]
As much as I want to write this up, I've been suffering from some sudden RSI and think it'd be better if I didn't. At least not until I see a doctor. Might be able to pick it up, but for now it's going on my back burner. Sorry! 🙈

**Comment by valscion** [comment]
That's entirely understandable @jmeas ❤️  no hard feelings! I understand RSI 😅 

![20170403_102441-1](https://cloud.githubusercontent.com/assets/482561/24599268/00333e3a-1858-11e7-86fa-386f88008d61.jpg)

**Comment by Kumarsammy** [comment]
Weird stuff.
Somehow my js parsed size is double than that of stat size. Is it okay @valscion ?

**Comment by jmeas** [comment]
My RSI has chilled out, so I'm gonna PR this this weekend!

**Comment by chenop** [comment]
My stat size is less the the parsed size - does it make sense?
I expected after minification to get a smaller size...

**Comment by valscion** [comment]
Open a new issue with more details if you think something is wrong :)

**Comment by Fewwy** [comment]
I want to understand the Stat-Parsed-Gzip behavior. What should be bigger than others? 
For example, I have a stat size of 7.52 MB, parsed size of 13.04 MB, and gzip of 2.04 MB. 
So? My understanding is that I should look at the "stat" first and if it's too big - try to make it smaller by using webpack tools. Am I correct?

**Comment by Fewwy** [comment]
@valscion Yeah, I think there is an issue with one of the libraries that I use. It imports too many icons that we're not utilizing in the UI. But I managed to optimize the bundle size using the Terser webpack plugin. My parsed bundle 13 MB became 5 :), gzipped reduced to 1.3 MB(previously 2 MB).
